File: idscp2/src/main/java/de/fhg/aisec/ids/idscp2/idscp_core/finite_state_machine/Transition.java
package de.fhg.aisec.ids.idscp2.idscp_core.finite_state_machine;

import java.util.function.Function;

/**
 * Transition class for State machine, provides a doTransition method
 * that returns the next state for a given event
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
class Transition {
    private Function<Event,State> eventHandler;

    Transition(Function<Event,State> eventHandler){
        this.eventHandler = eventHandler;
    }

    State doTransition(Event e){
        return eventHandler.apply(e);
    }
}

File: idscp2/src/main/kotlin/de/fhg/aisec/ids/idscp2/idscp_core/fsm/Transition.kt
package de.fhg.aisec.ids.idscp2.idscp_core.fsm

/**
 * Transition class for State machine, provides a doTransition method
 * that returns the fsm result containing of the next state for a given event and the result code
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
class Transition(private val eventHandler: (Event) -> FSM.FsmResult) {
    fun doTransition(e: Event): FSM.FsmResult {
        return eventHandler(e)
    }
}